File: Project/MyProject/MyProject/Source/MyProject/Private/BasicFrame/DelayHandle/DelayPriorityHandleMgr.cpp
#include "MyProject.h"
#include "DelayPriorityHandleMgr.h"

MY_BEGIN_NAMESPACE(MyNS)

DelayPriorityHandleMgr::DelayPriorityHandleMgr()
{
	this.mDeferredAddQueue = new NoPriorityList();
	this.mDeferredAddQueue.setIsSpeedUpFind(true);
	this.mDeferredDelQueue = new NoPriorityList();
	this.mDeferredDelQueue.setIsSpeedUpFind(true);

	this.mNoOrPriorityList = new PriorityList();
	this.mNoOrPriorityList.setIsSpeedUpFind(true);
	this.mNoOrPriorityList.setIsOpKeepSort(true);
}

void DelayPriorityHandleMgr::init()
{
	Super::init();
}

void DelayPriorityHandleMgr::dispose()
{
	if (null != this.mDeferredAddQueue)
	{
		this.mDeferredAddQueue.clear();
		this.mDeferredAddQueue = null;
	}
	if (null != this.mDeferredDelQueue)
	{
		this.mDeferredDelQueue.clear();
		this.mDeferredDelQueue = null;
	}
	if (null != this.mNoOrPriorityList)
	{
		this.mNoOrPriorityList.clear();
		this.mNoOrPriorityList = null;
	}

	Super::dispose();
}

void DelayPriorityHandleMgr::addPriorityObject(INoOrPriorityObject priorityObject, float priority)
{
	this.addNoOrPriorityObject(priorityObject, priority);
}

void DelayPriorityHandleMgr::removePriorityObject(ITickedObject tickObj)
{
	this.removeNoOrPriorityObject(tickObj);
}

MY_END_NAMESPACE

File: Project/MyProject/MyProject/Content/MyAsset/Lua/MyLua/Libs/DelayHandle/DelayPriorityHandleMgr.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.Core.GObject");

MLoader("MyLua.Libs.DataStruct.NoOrPriorityList.NoPriorityList.NoPriorityList");
MLoader("MyLua.Libs.DelayHandle.DelayNoOrPriorityHandleMgr");

-- 延迟优先级处理管理器
local M = GlobalNS.Class(GlobalNS.DelayNoOrPriorityHandleMgr);
M.clsName = "DelayPriorityHandleMgr";
GlobalNS[M.clsName] = M;

function M:ctor()
	self.mDeferredAddQueue = GlobalNS.new(GlobalNS.NoPriorityList);
	self.mDeferredAddQueue:setIsSpeedUpFind(true);
	self.mDeferredDelQueue = GlobalNS.new(GlobalNS.NoPriorityList);
	self.mDeferredDelQueue:setIsSpeedUpFind(true);

	self.mNoOrPriorityList = GlobalNS.new(GlobalNS.PriorityList);
	self.mNoOrPriorityList:setIsSpeedUpFind(true);
	self.mNoOrPriorityList:setIsOpKeepSort(true);
end

function M:init()
	M.super.init(self);
end

function M:dispose()
	M.super.dispose(self);
end

function M:addPriorityObject(priorityObject, priority)
	self:addNoOrPriorityObject(priorityObject, priority);
end

function M:removePriorityObject(tickObj)
	self:removeNoOrPriorityObject(tickObj);
end

return M;